src/Tokenizer/Token.php

```diff
@@ -43,6 +43,13 @@ final class Token
      */
     private ?int $id;
 
+    /**
+     * Token ID (if available) or token content otherwise.
+     *
+     * @var _PhpTokenKind
+     */
+    private $kind;
```

Comment:
When this property would be public (readonly), we could check token kinds without any function/method calls: `\T_CLASS === $token->kind`
Even for array of kinds, thanks to `IN_ARRAY` opcode:
`in_array($token->kind, [\T_CLASS, ';'], true)`

But not sure if it is worth it (performance wise) and if we want to go that path.
It could remove many many method calls (`isGivenKind`/`equals`).